packages/roosterjs-editor-plugins/lib/ContentEdit/features/markdownFeatures.ts

```diff
@@ -0,0 +1,127 @@
+import { PluginKeyboardEvent } from 'roosterjs-editor-types';
+import {
+    Browser,
+    toggleBold,
+    toggleItalic,
+    toggleUnderline,
+    toggleStrikethrough,
+} from 'roosterjs/lib';
+import { Editor, ContentEditFeature, Keys } from 'roosterjs-editor-core';
+
+const findTriggerCharactersAndCheckBetween = (triggerChar: string) => (
+    event: PluginKeyboardEvent,
+    editor: Editor
+) => {
+    let selObj = window.getSelection();
+    let anchorNode = selObj.anchorNode;
+    let genericTriggerCharacterIndex = anchorNode.textContent.lastIndexOf(' ' + triggerChar);
+    let alterationRequired = false;
+    let stringAdjust = 2;
+
+    if (
+        anchorNode.textContent.lastIndexOf(triggerChar) == 0 ||
+        anchorNode.textContent.lastIndexOf(triggerChar) == 1
+    ) {
+        genericTriggerCharacterIndex = anchorNode.textContent.lastIndexOf(triggerChar);
+    } else if (anchorNode.textContent.lastIndexOf(' ' + triggerChar) >= 0) {
+        genericTriggerCharacterIndex = anchorNode.textContent.lastIndexOf(' ' + triggerChar);
+    }
+
+    let firstWhiteSpaceChecker = anchorNode.textContent.charAt(
+        genericTriggerCharacterIndex + stringAdjust
+    );
+    let secondWhiteSpaceChecker = anchorNode.textContent.charAt(anchorNode.textContent.length - 1);
+    if (!(secondWhiteSpaceChecker.trim() || firstWhiteSpaceChecker.trim())) {
+        alterationRequired = true;
+    }
+
+    return alterationRequired;
+};
+
+const applyStyle = (triggerChar: string, generatedElement: string) => (
+    event: PluginKeyboardEvent,
+    editor: Editor
+) => {
+    let selObj = window.getSelection();
+    let anchorNode = selObj.anchorNode;
+    let stringToBeAltered = new Range();
+    let genericTriggerCharacterRange = new Range();
+    let stringAdjust = 1;
+    let scenario = 'N/A';
+    let genericTriggerCharacterIndex;
+
+    if (
+        anchorNode.textContent.lastIndexOf(triggerChar) == 0 ||
+        anchorNode.textContent.lastIndexOf(triggerChar) == 1
+    ) {
+        genericTriggerCharacterIndex = anchorNode.textContent.lastIndexOf(triggerChar);
+        scenario = 'begin';
+    } else if (anchorNode.textContent.lastIndexOf(' ' + triggerChar) >= 0) {
+        genericTriggerCharacterIndex = anchorNode.textContent.lastIndexOf(' ' + triggerChar);
+        scenario = 'mid';
+    }
+
+    if (scenario == 'begin') {
+        stringToBeAltered.setStart(anchorNode, genericTriggerCharacterIndex + stringAdjust);
+        stringToBeAltered.setEnd(anchorNode, anchorNode.textContent.length);
+
+        genericTriggerCharacterRange.setStart(anchorNode, genericTriggerCharacterIndex);
```

Comment:
`anchorNode.textContent.replaceAll` might be simpler here and below

Reply:
I'm getting "Property 'replaceAll' does not exist on type 'string'."

Reply:
whoops, `replaceAll` isn't in all browsers: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replaceAll

Reply:
`replace` should work just as well (replace and replace all both do the same thing pretty much)

Reply:
you could use `slice` here alternatively, or leave as is. not a big deal either way, just think it's cleaner than making a range and deleting its contents